Mathlib/Algebra/Polynomial/SumIteratedDerivative.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.BigOperators
+import Mathlib.Algebra.Polynomial.Degree.Lemmas
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Algebra.Polynomial.RingDivision
+
+/-!
+# Sum of iterated derivatives
+-/
+
+open Finset
+open scoped Nat
+
+namespace Polynomial
+
+variable {R : Type*}
+
+section Semiring
+
+variable [Semiring R]
+
+theorem sum_ideriv_apply_of_lt' {p : R[X]} {n : ℕ} (hn : p.natDegree < n) :
+    ∑ i in range (p.natDegree + 1), derivative^[i] p = ∑ i in range n, derivative^[i] p := by
+  obtain ⟨m, rfl⟩ := Nat.exists_eq_add_of_lt hn
+  rw [add_right_comm, sum_range_add _ _ m]
+  convert (add_zero _).symm
+  apply sum_eq_zero
+  intro x _
+  rw [add_comm, Function.iterate_add_apply, iterate_derivative_eq_zero (lt_add_one _),
+    iterate_derivative_zero]
+
+theorem sum_ideriv_apply_of_le' {p : R[X]} {n : ℕ} (hn : p.natDegree ≤ n) :
+    ∑ i in range (p.natDegree + 1), derivative^[i] p = ∑ i in range (n + 1), derivative^[i] p :=
+  sum_ideriv_apply_of_lt' (Nat.lt_add_one_iff.mpr hn)
+
+/-- Sum of iterated derivatives of a polynomial, as a linear map -/
+noncomputable def sumIderiv : R[X] →ₗ[R] R[X] where
```

Comment:
Should this be a weighted sum instead of a plain sum? A sentence or two explaining why this definition is useful (and/or more useful than the unweighted one) would be great in the module or function docstring.

Reply:
Only because it isn't needed so far; unless you have use cases to inform the design and what would be useful to prove about it, I think I'd prefer keeping it simple for now